app/services/publishers/vacancies/vacancy_step_process.rb

```diff
@@ -32,6 +42,24 @@ def previous_step_or_review
 
   private
 
+  def validate_step(step_name)
```

Comment:
I have a slight question mark about whether validation should really be the job of the VacancyStepProcess, whose main responsibility has been to house the logic that tracks which step should be rendered next. I'm open to it though.

Reply:
I'm not married to it being the job of the step process (though I'm fine with it), so long as it's not scattered around all the controllers and such.

Reply:
Is 'put the validation bits into a module/class and include the module' a good or a bad pattern (here, and in general)? It has the advantage of separating the concerns (_somewhat_).

From the look of it, `all_steps_valid?` and `validate_all_steps?` know nothing about vacancies, so they could be taken out, and/or live in parent class `StepProcess`.

Reply:
Putting things in modules for inclusion is good for composing functionality in different combinations, but since we're not doing that here I don't think it achieves much other than hide the code away.

I'm seeing the step process as a model that wraps the various form models, but perhaps that can be split out?

Reply:
I've created a new model to handle this.

Reply:
I like it! Thanks!